File: single-language-analysis/dataset-files/python/test_smell_104.py
# Original URL: https://github.com/OphisMusic/ophis/blob/afd3f6ab9aab9de57eaf633caaf24d0e7b5c74ad/tests/test_tonus.py#L132-L142

# Takes a long time... creates 6545 combinations
def test_smell_104():
    for x, y, z in itertools.combinations(ophis.western_chroma_set, 3):
        s = ophis.ChromaSet({x, y, z})
        s_dim = s.diminish()
        for chroma in s:
            assert chroma.diminish() in s_dim
        #for i in range(12):
        #    s_dim = s.diminish(i)
        #    for chroma in s:
        #        assert chroma.diminish(i) in s_dim

File: single-language-analysis/dataset-files/java/TestSmell104.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.Iterator;

public class TestSmell104 {

    @Test
    public void testDiminishChromaSet() {
        Iterator<Combination> combinations = new Combinations(ophis.westernChromaSet, 3).iterator();
        while (combinations.hasNext()) {
            Combination combination = combinations.next();
            ChromaSet s = new ChromaSet(Set.of(combination.getX(), combination.getY(), combination.getZ()));
            ChromaSet sDim = s.diminish();
            for (Chroma chroma : s) {
                assertTrue(sDim.contains(chroma.diminish()));
            }
        }
    }
}